components/director/cmd/systemfetcher/main.go

```diff
@@ -215,7 +213,26 @@ func createSystemFetcher(cfg config, cfgProvider *configprovider.Provider, tx pe
 	appTemplateRepo := apptemplate.NewRepository(appTemplateConv)
 	appTemplateSvc := apptemplate.NewService(appTemplateRepo, webhookRepo, uidSvc)
 
-	systemsAPIClient := systemfetcher.NewClient(cfg.APIConfig, cfg.OAuth2Config, systemfetcher.DefaultClientCreator)
+	var authProvider httputil.AuthorizationProvider
+
+	if cfg.OAuth2Config.ClientSecret == "" {
```

Comment:
This option to use mtls client is a bit hidden. Perhaps introducing a new option specifically for mtls would be better and more meaningful

Reply:
In the near future(~ 1 month) the system fetcher will use only the mTLS flow. Having that in mind, I thought it would be easier to delete the not needed code.